server/src/server.ts

```diff
@@ -129,21 +144,41 @@ async function getWorkspaceFolder(
   return undefined;
 }
 
-async function determineConfigPath(
-  uri: string,
-  textDocument: TextDocument,
-): Promise<string | null> {
-  const workspaceFolder: string | undefined =
-    await getWorkspaceFolder(textDocument);
+function getVSCodeOptions(
+  settings: ConfigurationSettings,
+  workspaceFolder: string | undefined,
+): Partial<ConfigurationSettings> | null {
+  if (!workspaceFolder) {
+    return null;
+  }
+
+  // remove configurationFilePath
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { configurationFilePath, ...restConfiguration } = settings;
+
+  // translate null (that means unsupecified option) to undefined
+  const removedNullSettings = Object.fromEntries(
+    Object.entries(restConfiguration).map(([key, value]) => [
```

Comment:
`undefined`に置き換えてもkey自体は除外されないので、filter使う方が良さそうです。

```suggestion
    Object.entries(restConfiguration).filter(([, value]) => value !== null)
```